server/websocket/src/storage/redis/mod.rs

```diff
@@ -531,4 +531,48 @@ impl RedisStore {
 
         Ok(updates)
     }
+
+    pub async fn acquire_oid_lock(&self, ttl_seconds: u64) -> Result<String, anyhow::Error> {
+        let lock_key = "lock:oid_generation";
+        let lock_value = uuid::Uuid::new_v4().to_string();
+        let mut conn = self.pool.get().await?;
+
+        let result: Option<String> = redis::cmd("SET")
+            .arg(lock_key)
+            .arg(&lock_value)
+            .arg("NX")
+            .arg("EX")
+            .arg(ttl_seconds)
+            .query_async(&mut *conn)
+            .await?;
+
+        if result.is_some() {
+            Ok(lock_value)
+        } else {
+            Err(anyhow::anyhow!("Failed to acquire OID generation lock"))
+        }
+    }
```

Comment:
Either use Lua script or used Redis Command among all the functions. not a mix of both